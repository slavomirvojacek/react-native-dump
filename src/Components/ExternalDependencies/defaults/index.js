import { noop } from '@openmaths/utils'

const props = {
  logIntoConsole: noop,
}

export {
  props
}
