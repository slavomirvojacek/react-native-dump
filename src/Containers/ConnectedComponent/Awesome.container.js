import React from 'react'
import { connect } from 'react-redux'

import Component from './Awesome.component'
import { apiFetch } from './lib/saga'

const mapStateToProps = ({AwesomeState}) => ({
  username: AwesomeState.data
    .map(_ => _.user.name),
  error: AwesomeState.error,
  showLoading: AwesomeState.isPending
})

const mapDispatchToProps = {
  apiFetch
}

export default connect(mapStateToProps, mapDispatchToProps)(Component)
