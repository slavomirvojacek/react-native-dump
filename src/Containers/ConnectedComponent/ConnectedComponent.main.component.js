import { throwIfNotBoolean, throwIfNotOption, throwIfNotFunction } from '@openmaths/utils'
import React, { Component } from 'react'
import { View, Text } from 'react-native'

import styles from './ConnectedComponent.main.component.styles'

const Loading = () => <Text>Loading...</Text>
const Error = e => <Text style={styles.error}>{e}</Text>
const Username = u => <Text style={styles.username}>{u.unwrap_or('N/A')}</Text>

class AwesomeComponent extends Component {
  static displayName = 'ConnectedComponent'

  componentDidMount () {
    const { apiFetch } = this.props

    throwIfNotFunction(apiFetch, '`apiFetch` has to be a Function', TypeError)

    apiFetch('John Doe')
  }

  render () {
    const {error, username, showLoading} = this.props

    throwIfNotBoolean(showLoading)
    throwIfNotOption(username)
    throwIfNotOption(error)

    return (
      <View>
        {showLoading
          ? Loading : error.match({
            some: Error,
            none: Username(username),
          })}
      </View>
    )
  }
}

export default AwesomeComponent
