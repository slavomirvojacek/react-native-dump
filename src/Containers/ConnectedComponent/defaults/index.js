import { None } from '@threestup/monads'

const props = {
  showLoading: false,
  username: None,
  error: None,

}

export {
  props
}
