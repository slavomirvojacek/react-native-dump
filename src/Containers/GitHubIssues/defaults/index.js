import { None } from '@threestup/monads'

const userDefault = (id = '') => ({id})

const reactionDefault = (override = {}) => {
  const def = {
    id: '',
    content: '',
    user: userDefault()
  }

  return {...def, ...override}
}

const issueDefault = (override = {}) => {
  const def = {
    id: '',
    title: '',
    url: '',
    bodyText: '',
    reactions: []
  }

  return {...def, ...override}
}

const props = {
  issuesLoading: false,
  issues: None,
  error: None,
}

export {
  userDefault, reactionDefault, issueDefault, props
}
