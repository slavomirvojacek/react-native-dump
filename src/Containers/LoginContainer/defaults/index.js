import { noop } from '@openmaths/utils'

import { initialState } from '../lib/reducer'

const props = {
  signIn: noop,
  loginState: initialState,
}

export {
  props
}
