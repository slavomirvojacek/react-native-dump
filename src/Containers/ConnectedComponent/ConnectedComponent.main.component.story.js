import { Some } from '@threestup/monads'
import React from 'react'
import { storiesOf } from '@storybook/react-native'

import { props as testDefaultProps } from './defaults'
import AwesomeComponent from './ConnectedComponent.main.component'

storiesOf('AwesomeComponent', module)
  .add('default', () => (
    <AwesomeComponent {...testDefaultProps} />
  ))
  .add('loading', () => {
    const props = {
      ...testDefaultProps,
      showLoading: true
    }
    return <AwesomeComponent {...props} />
  })
  .add('error', () => {
    const props = {
      ...testDefaultProps,
      error: Some('Err!')
    }
    return <AwesomeComponent {...props} />
  })
  .add('data present', () => {
    const props = {
      ...testDefaultProps,
      username: Some('Some User')
    }
    return <AwesomeComponent {...props} />
  })
