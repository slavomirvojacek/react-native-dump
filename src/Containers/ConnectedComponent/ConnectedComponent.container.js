import React from 'react'
import { connect } from 'react-redux'

import ConnectedComponent from './ConnectedComponent.main.component'
import { apiFetch } from './lib/saga'

const mapStateToProps = ({AwesomeState}) => ({
  username: AwesomeState.data
    .map(_ => _.user.name),
  error: AwesomeState.error,
  showLoading: AwesomeState.isPending
})

const mapDispatchToProps = {
  apiFetch
}

export default connect(mapStateToProps, mapDispatchToProps)(ConnectedComponent)
