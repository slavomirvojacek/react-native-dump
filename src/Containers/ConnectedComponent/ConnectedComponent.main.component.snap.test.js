import React from 'react'
import { Some, None } from '@threestup/monads'

import { props as testDefaultProps } from './defaults'
import ConnectedComponent from './ConnectedComponent.main.component'
import { assertSnapshots } from '../../TestUtils/snapshot'

jest
  .mock('./ConnectedComponent.main.component.styles', () => ({
    error: {},
    username: {},
  }))

describe('ConnectedComponent Snapshots', () => {
  const config = [
    {
      props: {...testDefaultProps},
      desc: 'renders correctly'
    },
    {
      props: {
        ...testDefaultProps,
        showLoading: true
      },
      desc: 'renders correctly when loading'
    },
    {
      props: {
        ...testDefaultProps,
        username: Some('John Doe'),
        showLoading: false
      },
      desc: 'renders correctly when not loading and username present'
    },
    {
      props: {
        ...testDefaultProps,
        username: None,
        showLoading: false
      },
      desc: 'renders correctly when not loading and username not present'
    },
    {
      props: {
        ...testDefaultProps,
        error: Some('Err!'),
        showLoading: false
      },
      desc: 'renders correctly when not loading and error present'
    }
  ]

  assertSnapshots(ConnectedComponent, config)
})
