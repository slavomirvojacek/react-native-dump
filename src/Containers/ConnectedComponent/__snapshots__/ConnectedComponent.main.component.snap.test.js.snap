// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedComponent Snapshots renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={Object {}}
  >
    N/A
  </Text>
</View>
`;

exports[`ConnectedComponent Snapshots renders correctly when loading 1`] = `<View />`;

exports[`ConnectedComponent Snapshots renders correctly when not loading and error present 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={Object {}}
  >
    Err!
  </Text>
</View>
`;

exports[`ConnectedComponent Snapshots renders correctly when not loading and username not present 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={Object {}}
  >
    N/A
  </Text>
</View>
`;

exports[`ConnectedComponent Snapshots renders correctly when not loading and username present 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={Object {}}
  >
    John Doe
  </Text>
</View>
`;
